app/controllers/concerns/course/users_controller_management_concern.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+module Course::UsersControllerManagementConcern
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :authorize_show!, only: [:students, :staff, :requests, :invitations]
+    before_action :authorize_edit!, only: [:update, :destroy]
+  end
+
+  def update # :nodoc:
+    if update_course_user(course_user_params)
+      success = t('course.users.update.success', role: t("course.users.role.#{@course_user.role}"))
```

Comment:
I think we should use a gem or something for the enum translation,  it's a bit pain that we need to manually construction the keys for all the enums. What do you think ?  If you agree I will submit a issue and we can fix that later.